src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -1205,15 +1205,15 @@ else if (future.isDone()) {
      */
     @Override
     public Message request(String subject, byte[] body, Duration timeout) throws InterruptedException {
-        return requestInternal(subject, null, body, timeout, cancelAction, true);
+        return requestInternal(subject, null, body, timeout, cancelAction, true, true);

```

Comment:
All `public Message request(...)` are a user made request. The original behavior intended for the flush to happen after request hit the socket, hence the true.